src/x509/verify.ts

```diff
@@ -192,6 +194,25 @@ class CertificateChainVerifier {
         throw new VerificationError('Incorrect certificate name chaining');
       }
     }
+
+    // Check pathlength constraints
+    for (let i = 0; i < path.length; i++) {
+      const cert = path[i];
+
+      // If the certificate is a CA, check the path length
+      if (cert.extBasicConstraints?.isCA) {
+        const pathLength = cert.extBasicConstraints.pathLenConstraint;
+
+        // The path length, if set, indicates how many intermediate
+        // certificates (NOT including the leaf) are allowed to follow. The
+        // pathLength constraint of any intermediate CA certificate MUST be
+        // greater than or equal to it's own depth in the chain (with an
+        // adjustment for the leaf certificate)
+        if (pathLength !== undefined && pathLength < i - 1) {
```

Comment:
Is the `pathLenConstraint` attribute always set on CA cert objects and when not explicitly it's`undefined`?

Is the path length set on the intermediate and the root? What are the path length constraints set to on the fulcio certs? 

Just walking through the logic in my head, seems like `i - 1` will be `0` for the intermediate and `1` for the root. So the path length constraint needs to be `<= -1` to fail for the intermediate (`>= 0` to succeed), and `<= 0` to fail for the root (`>= 1` to succeed).

Reply:
https://www.rfc-editor.org/rfc/rfc5280#section-4.2.1.9